tests/testthat/test-shinytest2_filteredData_DF.R

```diff
@@ -0,0 +1,231 @@
+      )),
+      "Filter Data"
+    )
+  })
+
+  it("filters for a dataset", {
+    app_driver <- local_app_driver()
+
+    selector_collapsable <- get_class(".accordion-item:nth-of-type(1) > div.accordion-collapse")
+    out <- app_driver$get_js(selector_collapsable)
+    testthat::expect_length(out, 4L)
+    testthat::expect_true("show" %in% unlist(out[[3L]]))
+
+    selector <- "#iris * button"
+    app_driver$click(selector = selector)
+    app_driver$wait_for_idle()
+
+    out <- app_driver$get_js(selector_collapsable)
+    testthat::expect_length(out, 4L)
+    testthat::expect_false("show" %in% unlist(out[[3L]]))
+
+    # The text is still in DOM
+    text <- app_driver$get_text("#filter_panel-active-iris-filter-iris_Species-inputs-selection")
+    testthat::expect_equal(
+      clean_text(text),
+      c("setosa (50/50)", "versicolor (50/50)", "virginica (50/50)")
+    )
+  })
+})
+
+testthat::describe("Clicking remove buttons removes the visibility of", {
+  ns <- function(dataset, id) shiny::NS(sprintf("filter_panel-active-%s", dataset), id)
+  id_ns <- function(dataset, id) sprintf("#%s", ns(dataset, id))
+
+  it("one filter", {
+    app_driver <- local_app_driver()
+    testthat::expect_true(is_visible(app_driver, id_ns("mtcars", "filter-4_cyl")))
+    app_driver$click(ns("mtcars", "filter-4_cyl-remove"))
+    app_driver$wait_for_idle()
+    testthat::expect_false(is_visible(app_driver, id_ns("mtcars", "filter-4_cyl")))
+  })
+
+  it("filters from a dataset", {
+    app_driver <- local_app_driver()
+    testthat::expect_true(is_visible(app_driver, id_ns("iris", "filter-iris_Species")))
+    testthat::expect_true(is_visible(app_driver, id_ns("mtcars", "filter-4_cyl")))
+    testthat::expect_true(is_visible(app_driver, id_ns("mtcars", "filter-mtcars_mpg")))
+    app_driver$click(ns("mtcars", "remove_filters"))
+    app_driver$wait_for_idle()
+    testthat::expect_true(is_visible(app_driver, id_ns("iris", "filter-iris_Species")))
+    testthat::expect_false(is_visible(app_driver, id_ns("mtcars", "filter-4_cyl")))
+    testthat::expect_false(is_visible(app_driver, id_ns("mtcars", "filter-mtcars_mpg")))
+  })
+
+  it("filters from all datasets", {
+    app_driver <- local_app_driver()
+    testthat::expect_true(is_visible(app_driver, id_ns("iris", "filter-iris_Species")))
+    testthat::expect_true(is_visible(app_driver, id_ns("mtcars", "filter-4_cyl")))
+    testthat::expect_true(is_visible(app_driver, id_ns("mtcars", "filter-mtcars_mpg")))
+    app_driver$click("filter_panel-active-remove_all_filters")
+    app_driver$wait_for_idle()
+    testthat::expect_false(is_visible(app_driver, id_ns("iris", "filter-iris_Species")))
+    testthat::expect_false(is_visible(app_driver, id_ns("mtcars", "filter-4_cyl")))
+    testthat::expect_false(is_visible(app_driver, id_ns("mtcars", "filter-mtcars_mpg")))
+  })
+})
+
+testthat::describe("exclude_varnames", {
+  it("locks editing the filter if set by the app developer", {
```

Comment:
exlude varnames doesn't lock editing, it (only) doesn't allow to add this filter using add-filter-variable dropdown. 
This test here checks if the `FilterStateExpr` has a lock icon.